Amazon.QLDB.Driver.Tests/AsyncQldbDriverTests.cs

```diff
@@ -240,93 +264,92 @@ public async Task TestAsyncExecuteWithFuncLambdaAndRetryPolicyReturnsFuncOutput(
-                {
-                    FirstPage = new Page
-                    {
-                        NextPageToken = null
-                    }
-                },
-                ResponseMetadata = new ResponseMetadata
-                {
-                    RequestId = "testId"
-                }
-            };
-        
-            var sendCommandResponseCommit = new SendCommandResponse
+            h1 = Transaction.Dot(h1, statement, new List<IIonValue> { });
+
+            mockClient.QueueResponse(startSessionResponse);
+            mockClient.QueueResponse(startTransactionResponse);
+            foreach (var ex in exceptions)
             {
-                CommitTransaction = new CommitTransactionResult
-                {
-                    TransactionId = TestTransactionId,
-                    CommitDigest = new MemoryStream(h1.Hash),
-                },
-                ResponseMetadata = new ResponseMetadata
+                mockClient.QueueResponse(ex);
+
+                // OccConflictException reuses the session so no need for another start session.
+                if (ex is not OccConflictException)
                 {
-                    RequestId = "testId"
+                    mockClient.QueueResponse(startSessionResponse);
                 }
-            };
-            var mockCreator = new Mock<Func<CancellationToken, Task<Session>>>();
-            var mockSession = new Mock<Session>(null, null, null, null, null);
-        
-            mockSession.Setup(x => x.StartTransactionAsync(default)).ReturnsAsync(sendCommandResponseWithStartSession.StartTransaction);
-            mockSession.SetupSequence(x => x.ExecuteStatementAsync(It.IsAny<String>(), It.IsAny<String>(), It.IsAny<List<IIonValue>>(), default))
-                .Throws(exception)
-                .Throws(exception)
-                .ReturnsAsync(sendCommandResponseExecute.ExecuteStatement);
-            mockSession.Setup(x => x.CommitTransactionAsync(It.IsAny<String>(), It.IsAny<MemoryStream>(), default))
-                .ReturnsAsync(sendCommandResponseCommit.CommitTransaction);
-        
-            mockCreator.Setup(x => x(default)).ReturnsAsync(mockSession.Object);
-        
-            var driver = new AsyncQldbDriver(
-                new AsyncSessionPool(mockCreator.Object, AsyncQldbDriverBuilder.CreateDefaultRetryHandler(NullLogger.Instance), 4, NullLogger.Instance));
-        
+
+                mockClient.QueueResponse(startTransactionResponse);
+            }
+            mockClient.QueueResponse(executeResponse(null));
+            mockClient.QueueResponse(commitResponse(h1.Hash));
+
+            var driver = new AsyncQldbDriver("ledgerName", mockClient, 4, NullLogger.Instance);
+
             try
             {
-                await driver.Execute(txn => txn.Execute(statement));
+                await driver.Execute(txn => txn.Execute(statement), policy);
+
+                Assert.IsNull(expectedExceptionType);
```

Comment:
`expectedExceptionType` is null when we don't expect Execute() to throw. And non-null when we do expect an exception.

Reply:
My last comment is moot now as I just reworked this so the test cases pass in an expectThrow flag instead of expectedExceptionType and then we infer the expected type by checking the type of the last exception in the list.